distributed/scheduler.py

```diff
@@ -758,12 +758,13 @@ def remove_worker(self, stream=None, address=None, safe=False):
             del self.worker_bytes[address]
 
             for key in self.has_what.pop(address):
-                self.who_has[key].remove(address)
-                if not self.who_has[key]:
-                    if key in self.tasks:
-                        recommendations[key] = 'released'
-                    else:
-                        recommendations[key] = 'forgotten'
+                if key in self.who_has:
```

Comment:
In `transition_memory_released`, there is the following:
```
            workers = self.who_has.pop(key)
            for w in workers:
                if w in self.worker_info:  # in case worker has died
                    self.has_what[w].remove(key)
```

I was seeing an ignored KeyError before the change.